genie-common-internal/src/main/java/com/netflix/genie/common/internal/dto/JobDirectoryManifest.java

```diff
@@ -0,0 +1,407 @@
+     *
+     * @return The total size (in bytes) of all the files in this job directory
+     */
+    @JsonIgnore
+    public long getTotalSizeOfFiles() {
+        return this.totalSizeOfFiles;
+    }
+
+    @Slf4j
+    private static class ManifestVisitor extends SimpleFileVisitor<Path> {
+
+        private final Path root;
+        private final ImmutableMap.Builder<String, ManifestEntry> builder;
+        private final TikaConfig tikaConfig;
+
+        ManifestVisitor(final Path root, final ImmutableMap.Builder<String, ManifestEntry> builder) throws IOException {
+            this.root = root;
+            this.builder = builder;
+            try {
+                this.tikaConfig = new TikaConfig();
+            } catch (final TikaException te) {
+                log.error("Unable to create Tika Configuration due to error", te);
+                throw new IOException(te);
+            }
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public FileVisitResult preVisitDirectory(final Path dir, final BasicFileAttributes attrs) throws IOException {
+            final ManifestEntry entry = this.buildEntry(dir, attrs, true);
+            log.debug("Created manifest entry for directory {}", entry);
+            this.builder.put(entry.getPath(), entry);
+            return FileVisitResult.CONTINUE;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public FileVisitResult visitFile(final Path file, final BasicFileAttributes attrs) throws IOException {
+            final ManifestEntry entry = this.buildEntry(file, attrs, false);
+            log.debug("Created manifest entry for file {}", entry);
+            this.builder.put(entry.getPath(), entry);
+            return FileVisitResult.CONTINUE;
+        }
+
+        private ManifestEntry buildEntry(
+            final Path entry,
+            final BasicFileAttributes attributes,
+            final boolean directory
+        ) throws IOException {
+            final String path = this.root.relativize(entry).toString();
+            final Instant lastModifiedTime = attributes.lastModifiedTime().toInstant();
+            final Instant lastAccessTime = attributes.lastAccessTime().toInstant();
+            final Instant creationTime = attributes.creationTime().toInstant();
+            final long size = attributes.size();
+
+            String md5 = null;
+            String mimeType = null;
+            if (!directory) {
+                try (InputStream data = Files.newInputStream(entry, StandardOpenOption.READ)) {
+                    md5 = DigestUtils.md5Hex(data);
+                } catch (final IOException ioe) {
+                    // For now MD5 isn't critical or required so we'll swallow errors here
+                    log.error("Unable to create MD5 for {} due to error", entry, ioe);
+                }
+
+                try (TikaInputStream inputStream = TikaInputStream.get(entry)) {
```

Comment:
Spring is currently doing file content type detection, is there no way to reuse their code, rather than this new dependency?
Just curious, if this works I don't care much what we end up using.